spec/message.abnf

```diff
@@ -52,16 +55,18 @@ unquoted-start = name-start / DIGIT / "."
                / %xB7 / %x300-36F / %x203F-2040
 
 
-; reserve sigils for private-use by implementations
-private-use    = private-start reserved-body
-private-start  = "^" / "&"
+; Reserve additional .keywords for use by future versions of this specification.
+; Note that the following expression is a simplification,
+; as this rule MUST NOT be considered to match existing keywords:
+;   (input / local / match) (s / "{")
+reserved-statement = "." reserved-body 1*expression
```

Comment:
@aphillips While working on the follow-up changes to this PR, I recalled that we'd discussed being more specific and limited about the syntax space that we reserve for statements, so that an actual keyword would always be required. Would this be an improvement in that regard?

```suggestion
reserved-statement = reserved-keyword [s reserved-body] 1*expression
; Note that the following expression is a simplification,
; as this rule MUST NOT be considered to match existing keywords.
reserved-keyword = "." 2*(%x61-7A)  ; (a-z)
```

Reply:
Yes, this looks good (with the `[s]` edit). I probably would not do `2*` even though we're unlikely to make a one letter keyword. I would use our new-found case sensitive ABNF as well and pull the regex comment above (keeping the other comment parts):

```abnf
; as regexp: /\.(?!input|local|match)[a-z]*
reserved-keyword = "." *(%s(a-z))
```

Reply:
Edited the suggestion to use `1*` instead of `2*`, and moved the comment.

Alas, `%s` syntax doesn't support ranges, so need to stick with the `%x` here.